test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,144 @@
+ */
+class cache_resize : public test_harness::test {
+    public:
+    cache_resize(const test_harness::test_args &args) : test(args)
+    {
+        delete this->_workload_tracking;
+        this->_workload_tracking =
+          new tracking_table_cache_resize(_config->get_subconfig(WORKLOAD_TRACKING),
+            _config->get_bool(COMPRESSION_ENABLED), *_timestamp_manager);
+    }
+
+    void
+    custom_operation(test_harness::thread_context *tc) override final
+    {
+        bool increase_cache = true;
+        const std::string small_cache_size("cache_size=1MB");
+        const std::string big_cache_size("cache_size=500MB");
+
+        uint64_t prev_cache_size;
+        while (tc->running()) {
+            tc->sleep();
+            WT_CONNECTION *conn = connection_manager::instance().get_connection();
+            prev_cache_size = ((WT_CONNECTION_IMPL *)conn)->cache_size;
+            testutil_check(conn->reconfigure(
+              conn, increase_cache ? big_cache_size.c_str() : small_cache_size.c_str()));
+            increase_cache = !increase_cache;
+            logger::log_msg(LOG_INFO,
+              "The cache size was updated from " + std::to_string(prev_cache_size) + " to " +
+                std::to_string(((WT_CONNECTION_IMPL *)conn)->cache_size));
+        }
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
+    {
+        uint64_t coll_id = 0;
+        collection &coll = tc->db.get_collection(coll_id);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+
+        while (tc->running()) {
+
+            tc->transaction.try_begin();
+
+            /* Insert a random value. */
+            const std::string value =
+              random_generator::instance().generate_pseudo_random_string(tc->value_size);
+            bool ret = tc->insert(cursor, coll.id, value);
+
+            if (!ret) {
+                tc->transaction.rollback();
+            } else if (tc->transaction.can_commit()) {
+                testutil_assert(tc->transaction.commit());
+            }
+
+            tc->sleep();
+        }
+
+        /* Make sure the last transaction is rolled back now the work is finished. */
+        if (tc->transaction.active())
+            tc->transaction.rollback();
+    }
+
+    void
+    validate(const std::string &operation_table_name, const std::string &,
+      const std::vector<uint64_t> &) override final
+    {
+        WT_DECL_RET;
+        uint64_t tracked_cache_size, tracked_txn_id;
+
+        scoped_session session = connection_manager::instance().create_session();
+        scoped_cursor cursor = session.open_scoped_cursor(operation_table_name);
+
+        uint64_t num_records = 0;
+        while ((ret = cursor->next(cursor.get())) == 0) {
+            testutil_check(cursor->get_value(cursor.get(), &tracked_cache_size, &tracked_txn_id));
+            /*
+             * Only transactions that were executed under a large enough cache should have been
```

Comment:
This could do with some more context. It's implicit knowledge that when an insert to the DB fails we don't add it to the tracking table.

Reply:
I agree, I added more context as part of 309deeb40fbc3d80a901d1fb5ea55f9f4feba97d